packages/emotion-primitives/src/props.js

```diff
@@ -0,0 +1,131 @@
+import viewStylePropTypes from 'react-primitives/lib/web/View/ViewStylePropTypes'
+import textStylePropTypes from 'react-primitives/lib/web/Text/TextStylePropTypes'
+import imageStylePropTypes from 'react-primitives/lib/web/Image/ImageStylePropTypes'
+
+// <View /> Style props
+const viewStyleProps = Object.keys(viewStylePropTypes)
+// <Text /> Style props
+const textStyleProps = Object.keys(textStylePropTypes)
+// <Image /> Style props
+const imageStyleProps = Object.keys(imageStylePropTypes)
+
+// <Text /> primitive props
+const textProps = [
```

Comment:
we could create RegExps out of those array at build time - like [here](https://github.com/emotion-js/next/blob/e4f914824774a47618d23427397c56c367c7bbe2/packages/is-prop-valid/src/index.js#L5), plus testing functions could be memoized to return results from the cache

Reply:
We should probably use the objects themselves and do a !== undefined check since that’s what the memoized version would do

Reply:
Do u remember why RegExp was introduced there in the first place? Purely as size optimization? If that's the case the same technique should be reused here or this build-time RegExp should get replaced in the other place too.

Reply:
It was mainly for size but also we need to check for some things that aren’t on a list like `on*`, `aria-*` and `data-*` and regex is the easiest way to do that so if we already have to do a regex check we might as well put everything there.

The size concern isn’t really here since we’re importing the object from another package so it will be included anyway and if we included regex it would be larger since there would be our regex + the prop type objects

Reply:
That's true about `viewStyleProps`, `textStyleProps`, `imageStyleProps`, but not for `textProps`, `viewProps` and `imageProps`, right? those could be optimized with a RegExp approach

Reply:
Yep, you’re right, those can be done with regex.

@nitin42 I was looking at styled-components' native and primitives implementation and I noticed they don’t do prop filtering and I ended up trying a random prop on react native and noticed there are no unknown prop warnings. Is prop filtering necessary? I’m guessing it is because they have to be filtered on the web? I just wanted to ask to make sure

Reply:
Hm... I'll try to reference this with my work on glamorous primitives. Doing this checks seems unnecessary to me, now!